### ISSUE TITLE
Critical and unbelievable bug makes impossible to deploy to IIS

### ISSUE THREAD

**Issue by shamnigurat** [Microsoft/vsts-rm-extensions#115]
I'll be brief.
1) Look closely at this:
Deploy-Website -webDeployPkg $WebDeployPackage -webDeployParamFile $WebDeployParamFile -overRiderParams $OverRideParams -websiteName $websiteName -excludeFilesFromAppData $excludeFilesFromAppData -removeAdditionalFiles $removeAdditionalFiles -takeAppOffline $takeAppOffline -additionalArguments $AdditionalArguments

There is the $websiteName parameter.

2) Look closely at this:
function Deploy-Website
{    
    param(
        [string]$webDeployPkg,
        [string]$webDeployParamFile,
        [string]$overRiderParams,
        [string]$removeAdditionalFiles,
        [string]$excludeFilesFromAppData,
        [string]$takeAppOffline,
        [string]$additionalArguments
    )

I'm intrested in what has happened with the $websiteName parameter, uh?

3) Look at the three lines above:
$msDeployCmdArgs = [string]::Format(' -verb:sync -source:package="{0}" {1} -dest:auto -retryAttempts:3 -retryInterval:3000', $webDeployPackage, $msDeployCmdArgs)

The curious question: HOW IS IT SUPPOSED TO DEPLOY TO IIS's SITE DEFINED BY $websiteName PARAMETER (the value comes from the UI-from of the 'WinRM - IIS Web App Deployment' task)???

I'm in shock of how it could be possible to be so handless clown to write such miserable code.
It's called 'the hindu code'.

Hints to fix:
1) parameter $websiteName should be passed down to the Get-MsDeployCmdArgs function;
2) value of the 'dest' parameter for msdeploy should be iisApp:"$websiteName" instead of 'auto'.

The last question: when will be this shameful bug fixed???

**Comment by mvvsubbu** [comment]
The parameter $websiteName is never used in Deploy-Website. Override parameters contains details about where to deploy. Unused parameter I will remove

**Comment by shamnigurat** [comment]
It's unacceptable because without fixing this issue as I mentioned ([#115](https://github.com/Microsoft/vsts-rm-extensions/issues/115)) deployment is not working at all.
Just look at the parameters of the "WinRM - IIS Web App Deployment" task.
There is the parameter named "Website Name". Just tell me: what is this parameter for?
The information's hint tells: "Provide the name of the website **for deploying the Web project to**."
But this parameter is not used while deploying to IIS so I can't deploy as hint tells me!
Further more, just imagine the case: I have zipped web-package (msdeploy) and want to deploy it to IIS. All I know is the name of the web site to deploy - how can I deploy in this case (I don't know the physical path of the web site and I don't need it because I would be able to deploy on differnet environments with the same site's name!)? The "Override Parameters" doesn't work for this purpose even if I set 'name="IIS Web Application Name",value="...."' - msdeploy just tries to unpack the content of the web-package to the physical path the package was created from.
Wihtout fixing my [issue](https://github.com/Microsoft/vsts-rm-extensions/issues/115) the task "WimRM - IIS Web App Deployment" becomes almost useless.
